module/executiondatasync/execution_data/execution_data_producer.go

```diff
@@ -0,0 +1,55 @@
+package execution_data
+
+import (
+	"errors"
+
+	"go.uber.org/atomic"
+
+	"github.com/onflow/flow-go/engine"
+)
+
+var ErrMultipleRegister = errors.New("producer may only be registered once")
+
+type ExecutionDataProducer interface {
+	Register(*engine.Notifier) error
+	SetLastProcessedHeight(uint64)
```

Comment:
We typically use names like `On*` for this type of use case. Maybe rename to something like

```suggestion
	OnBlockProcessed(uint64)
```

Reply:
[4b87f42](https://github.com/onflow/flow-go/pull/6217/commits/4b87f42a1f9693edeb7cfcc1c959532202400d71)